src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -855,6 +900,10 @@ public RuntimeType? ExposedClassObject
                 return *(RuntimeType*)Unsafe.AsPointer(ref ExposedClassObjectRaw);
             }
         }
+
+        public bool IsClassInited => (Volatile.Read(ref Flags) & enum_flag_Initialized) != 0;
```

Comment:
Is volatile read required? For thread safety, I assume the cctor logic should check for multiple entrance.

Reply:
Yes, the volatile read is does that check.